specs/anon-aadhaar/specs.md

```diff
@@ -0,0 +1,542 @@
+## Anon-Aadhaar spec raw
+
+---
+
+slug: CS-02
+title: CS-02/ANON-AADHAAR-V2
+name: Anonymous Aadhaar Verification Protocol
+status: raw
+category: Standards Track
+editor: [Yanis Meziane] <yanis@pse.dev>
+contributors:
+
+- [@Saleelp, @Oskarth]
+  tags:
+- zero-knowledge
+- identity
+- privacy
+
+---
+
+# Change Process
+
+This document is governed by the [1/COSS](https://rfc.vac.dev/spec/1/).
+
+# Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+# Abstract
+
+Anon Aadhaar is a zero-knowledge protocol enabling privacy-preserving verification of the Aadhaar Secure QR Code. The protocol allows proving possession of valid UIDAI-issued Aadhaar documents without revealing personal information, using RSA signature verification in zero-knowledge combined with selective disclosure mechanisms.
+
+# Motivation
+
+Current Aadhaar verification methods expose sensitive personal data. This specification provides:
+
+- Privacy-preserving verification of government-issued identity
+- Selective disclosure of specific attributes
+- Prevention of identity reuse through nullifiers
+- Decentralized verification without trusted intermediaries (e.g. verifying proof of a valid Aadhaar on the EVM)
+
+# Specification
+
+## System Requirements
+
+Implementations MUST provide:
+
+1. RSA-2048 signature verification
+2. SHA-256 hashing
+3. QR code data parsing
+4. Secure random number generation
+
+## Protocol Flow
+
+### 1. Document Processing
+
+Implementations MUST:
+
+1. Extract the signed message and the signature from the QR code data
+2. Verify RSA signature
+3. Parse signed data fields
+4. Generate circuit inputs
+
+Input requirements:
+
+```
+{
+  qrDataPadded: Bytes,        // REQUIRED - QR code padded signed data
+  qrDataPaddedLength: Bytes32 // REQUIRED - Length of the the QR code padded data
+  signature: Bytes,           // REQUIRED - RSA signature
+  pubKey: Bytes,              // REQUIRED - Public key
+  nullifierSeed: Bytes32,     // REQUIRED - Random seed of the nullifier
+  signalHash: Bytes32,        // OPTIONAL - Binding data
+  ageAbove18: Boolean,        // OPTIONAL - Disclosure flags
+  gender: Boolean,            // OPTIONAL - Disclosure flags
+  state: Boolean,             // OPTIONAL - Disclosure flags
+  pincode: Boolean            // OPTIONAL - Disclosure flags
+
+}
+```
+
+## Circuit Design
+
+### Inputs
+
+#### Private Inputs
+
+- `qrDataPadded: Bytes[]` - Padded signed data from QR code
+- `qrDataPaddedLength: BigInt` - Length of the the QR code padded data
```

Comment:
Should it explain what is meant by `BigInt`?

Reply:
Yes good point, maybe a brief reference in prelude of section or so

Reply:
This type of interface is linked to the proving system used, so I'm not sure about the right level of abstraction here. 
Should it just specify the format of the data to be processed, or match the exact data type used in the current reference implementation? 
What are your thoughts?